server/src/main/java/org/apache/cassandra/sidecar/config/ServiceConfiguration.java

```diff
@@ -179,4 +180,11 @@ default WorkerPoolConfiguration serverInternalWorkerPoolConfiguration()
      * @return the configuration relevant to the coordination functionality of Sidecar
      */
     CoordinationConfiguration coordinationConfiguration();
+
+    /**
+     * @return Sidecar's DNS Resolver to use. Default will be to provide forward and reverse DNS resolution which
+     * provides both the hostname & address. The alternative, resolveToIp, only provides forward resolution meaning
+     * the IP address will be returned for both hostname & address.
+     */
+    DnsResolver.DnsResolverEnum dnsResolver();
```

Comment:
Here we should use the interface instead.

Reply:
Updated